test/end-to-end/UrlCreation.test.ts

```diff
@@ -218,3 +229,56 @@ test('The URL searching test.', async (t) => {
     .expect(resultTable.child('tbody').child(0).child(1).child(0).child(0).child('h6').innerText)
     .eql(`/${generatedUrlActive}-search`)
 })
+
+test('The bulk based test.', async (t) => {
+  await t.click(createLinkButton.nth(0))
+
+  const longUrls = Array(100).fill('https://google.com')
+  const currLinkCount = await getLinkCount()
+  const expectedLinkCount = currLinkCount + longUrls.length
+
+  // valid file
+  await createBulkCsv(dummyBulkCsv, longUrls)
+
+  await t
+    .click(bulkTab)
+    .setFilesToUpload(uploadFile, dummyBulkCsvRelativePath)
+    .click(tagsAutocompleteInput)
+    .typeText(tagsAutocompleteInput, tagText3)
+    .pressKey('enter')
+    .click(createLinkButton.nth(2))
+
+  await t
+    // It should show an success snackbar when a new file link has been added
+    .expect(successBulkCreation.exists)
+    .ok()
+    // The number of links should increase by numLongUrls
+    .expect(await getLinkCount())
+    .eql(expectedLinkCount)
```

Comment:
for some reason, on my local environment `await getLinkCount()` still returns the old value before the 100 new links were created :/

it doesn't appear to be waiting for the number of links to update - but if I add e.g. an extra second of wait time, then it passes on local dev

Reply:
Interesting! Thanks for finding this bug - it works on my computer but I think we can add a small wait time anyway just preemptively to reduce flakiness

Reply:
added here [6987340](https://github.com/opengovsg/GoGovSG/pull/2022/commits/698734090f0c532493751b9023e7bbf71ec50754)